### PR TITLE
Sfrcleanup

### PR DESCRIPTION
**Pull request by sbird**
This is the first part of a series on removing thread contention in the star formation code. I split this one out to make the whole thing easier to review - this part is just cleanups. The main nice thing about it is that it removes the global thread-local variables in sfr_eff.c, instead just using locals with omp reduction clauses. Hooray! I also split winds into a separate file.

One thing I did was to make -DSFR just guard the wind model, so I could unify the routines for cooling and star formation and just cooling. This should prevent us from breaking compilation of the !SFR case. I also considered removing -DSFR entirely - the main runtime penalty is an extra double in SPHP which isn't really a big deal. What do you think?

### CODE REVIEW

**Comment by sbird** [comment]
Tested that it gives identical results for the hydro example

**Comment by rainwoodman** [comment]
Yes on removing the SFR conditional compilation flag.

File: libgadget/sfr_eff.c [hunk 1]
```diff
@@ -23,373 +23,118 @@
...
-    tw->postprocess = (TreeWalkProcessFunction) cool_postprocess;
-
-    treewalk_run(tw, ActiveParticle, NumActiveParticle);
-
-    walltime_measure("/Cooling/StarFormation");
-}
-
+#ifdef SFR
+    /* Now apply the wind model: has to use the new NumActiveParticle.*/
+    winds_and_feedback(ActiveParticle, NumActiveParticle);
```
Comment (rainwoodman) [inline libgadget/sfr_eff.c]: But NumActiveParticle is also a global variable?

File: libgadget/sfr_eff.c [hunk 2]
```diff
@@ -447,298 +192,68 @@ cooling_direct(int i) {
...
-        if(P[target].IsNewParticle && !P[target].DensityIterationDone) {
-             return 1;
-        }
-    }
-    return 0;
-}
-
-static int
-sfr_wind_feedback_haswork(int target, TreeWalk * tw)
+static int make_particle_star(int i, double mass_of_star)
```
Comment (rainwoodman) [inline libgadget/sfr_eff.c]: Please describe the meaning of retflag.

Comment (sbird) [reply libgadget/sfr_eff.c]: ok, will do. If it helps retflag disappears in the later (feature) commits

File: libgadget/sfr_eff.c (hunk 1 above)
Comment (sbird) [reply libgadget/sfr_eff.c]: Comment is badly worded: I mean that NumActiveParticle changes within the loop when new particles are spawned, so one cannot use nactive.

**Comment by sbird** [comment]
Ok! I'll take out -DSFR

**Pull request merged**